Find a functionally equivalent code:android.media.MediaFocusControl.unregisterMediaButtonIntent:COMMENT
Method Modifier: protected   
Comment:/**
 * see AudioManager.unregisterMediaButtonIntent(PendingIntent mediaIntent)
 * precondition: mediaIntent != null, eventReceiver != null
 */

@@ -1,13 +1,11 @@
 {
     Log.i(TAG, "  Remote Control   unregisterMediaButtonIntent() for " + mediaIntent);
-    synchronized (mAudioFocusLock) {
-        synchronized (mRCStack) {
-            boolean topOfStackWillChange = isCurrentRcController(mediaIntent);
-            removeMediaButtonReceiver_syncAfRcs(mediaIntent);
-            if (topOfStackWillChange) {
-                // current RC client will change, assume every type of info needs to be queried
-                checkUpdateRemoteControlDisplay_syncAfRcs(RC_INFO_ALL);
-            }
+    synchronized (mPRStack) {
+        boolean topOfStackWillChange = isCurrentRcController(mediaIntent);
+        removeMediaButtonReceiver_syncPrs(mediaIntent);
+        if (topOfStackWillChange) {
+            // current RC client will change, assume every type of info needs to be queried
+            checkUpdateRemoteControlDisplay_syncPrs(RC_INFO_ALL);
         }
     }
 }

